replica/database.cc

```diff
@@ -468,6 +468,48 @@ void backlog_controller::update_controller(float shares) {
     _scheduling_group.set_shares(shares);
 }
 
+constexpr backlog_controller::control_point compaction_controller::middle_point() {
+    return { 1.5, 100 };
+}
+constexpr backlog_controller::control_point compaction_controller::base_last_point() {
+    return { base_normalization_factor, 1000 };
+}
+
+const float compaction_controller::minimum_effective_max_shares() {
+    return middle_point().output * 2;
+}
+const float compaction_controller::default_max_shares() {
```

Comment:
But after max_shares is updated this will no longer return the default.
How about defining this as constexpr and let `base_last_point()` set `default_max_shares()` rather than a constant `1000`?

Reply:
Actually, it uses output of const "base" last point, which is immutable. So always 1k.